pkg/devicerepository/store/bleve/create.go

```diff
@@ -0,0 +1,186 @@
+	BrandID   string
+	BrandName string
+}
+
+type indexableModel struct {
+	BrandPB string // *ttnpb.EndDeviceBrand marshaled into JSON string
+	ModelPB string // *ttnpb.EndDeviceModel marshaled into JSON string
+
+	// Stored in separate fields to support ordering search results
+	BrandID   string
+	BrandName string
+	ModelID   string
+	ModelName string
+}
+
+const (
+	brandsIndexPath = "brandsIndex.bleve"
+	modelsIndexPath = "modelsIndex.bleve"
+)
+
+func newIndex(path string, overwrite bool, keywords ...string) (bleve.Index, error) {
+	mapping := bleve.NewIndexMapping()
+	if st, err := os.Stat(path); err == nil && st.IsDir() && overwrite {
+		if err := os.RemoveAll(path); err != nil {
+			return nil, err
+		}
+	}
+	keywordFieldMapping := bleve.NewTextFieldMapping()
+	keywordFieldMapping.Analyzer = keyword.Name
+	for _, keyword := range keywords {
+		mapping.DefaultMapping.AddFieldMappingsAt(keyword, keywordFieldMapping)
+	}
+	return bleve.New(path, mapping)
+}
+
+func getWorkingDirectory(paths []string) (string, error) {
+	for _, path := range paths {
+		if s, err := os.Stat(path); err == nil && s.IsDir() {
+			return path, nil
+		}
+	}
+
+	// Default to working directory
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(wd, "device-repository"), nil
+}
+
+// Initialize fetches the Device Repository package file and generates index files.
+func (c Config) Initialize(ctx context.Context, fetcher fetch.Interface, overwrite bool) error {
+	wd, err := getWorkingDirectory(c.SearchPaths)
+	if err != nil {
+		return err
+	}
+
+	b, err := fetcher.File("master.zip")
+	if err != nil {
+		return err
+	}
+
+	if err := unarchive(b, wd, func(path string) (string, bool) {
+		path = strings.TrimPrefix(path, "lorawan-devices-master/")
+		if !strings.HasPrefix(path, "vendor/") || (!strings.HasSuffix(path, ".yaml") && !strings.HasSuffix(path, ".js")) {
+			return "", true
+		}
+		return path, false
+	}); err != nil {
+		return err
+	}
+	s := remote.NewRemoteStore(fetch.FromFilesystem(wd))
+	brandsIndex, err := newIndex(path.Join(wd, brandsIndexPath), overwrite, "BrandID")
+	if err != nil {
+		return err
+	}
+	defer brandsIndex.Close()
+	modelsIndex, err := newIndex(path.Join(wd, modelsIndexPath), overwrite, "BrandID", "ModelID")
+	if err != nil {
+		return err
+	}
+	defer modelsIndex.Close()
+
+	brands, err := s.GetBrands(store.GetBrandsRequest{
+		Paths: ttnpb.EndDeviceBrandFieldPathsNested,
+	})
+	if err != nil {
+		return err
+	}
+
+	brandsBatch := brandsIndex.NewBatch()
+	modelsBatch := modelsIndex.NewBatch()
+	for _, brand := range brands.Brands {
+		log.FromContext(ctx).WithField("brand_id", brand.BrandID).Debug("Indexing")
+		models, err := s.GetModels(store.GetModelsRequest{
+			Paths:   ttnpb.EndDeviceModelFieldPathsNested,
+			BrandID: brand.BrandID,
+		})
+		if errors.IsNotFound(err) {
+			// Skip vendors without any models
+			continue
+		} else if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if err != nil {
			if errors.IsNotFound(err) {
				// Skip vendors without any models
				continue
			}
			return err
		}
```

Is faster and looks better